### DISCUSSION TITLE
openfeature provider?

### DISCUSSION THREAD

**Discussion by justinabrahms** [permitio/OPToggles#22] [Ideas]
Hi. I'm active w/ the openfeature project (part of the CNCF). https://openfeature.dev. It's a vendor-agnostic flagging SDK. I think OPToggles could be a natural fit for that ecosystem. There's a k8s aware flagging thing called flagD which may be a reasonable integration point. Happy to answer questions if you have them. We're in the #openfeature channel in the CNCF slack.

**Comment by orweis** [comment]
Hi @justinabrahms ! Thanks for reaching out  💜
Openfeature looks awesome, and I think it would makes a lot of sense to connect things here in a standard way.

I was able to clearly see how you use Openfeature to work with feature-toggling platforms as a source - i.e. use them in a standardized way to apply toggles to your code.

But, I wasn't able to find a way to work with them as a destination; i.e. load state into them to affect flags - which is what OPToggles does.
Anything you can point me at, or thoughts in general on the matter ?

**Reply by justinabrahms** [reply]
Yes! You're looking for the provider interface. It's a class with like 4 methods to implement then you're done.

**Comment by orweis** [comment]
Hi @justinabrahms .
I'm not sure we are on the same page here, or I'm misunderstand something :) 

I see providers have resolve methods (to resolve data coming from the flagging solution), but OPToggles would need a set/apply method to save flag values into the flagging-solution.

**Reply by justinabrahms** [reply]
Ahh! My misunderstanding. I thought y'all were a flagging provider based on opa rules.  If you're flag authoring place, you might integrate instead with https://github.com/open-feature/flagd , which is a cloud native flagging provider. Does that feel like a better fit?
